Src/Plugins/Security/SEC101_043.NuGetCredentialsValidator.cs

```diff
@@ -211,5 +214,48 @@ private static List<string> ExtractHostsHelper(string hostXmlAsString)
 
             return returnList ?? new List<string>();
         }
+
+        private static List<(string user, string password)> ExtractCredentials(string credentialXmlAsString)
+        {
+            var list = new List<(string user, string password)>();
+
+            try
+            {
+                var credentialsXml = new XmlDocument();
+                credentialsXml.LoadXml(credentialXmlAsString);
+
+                IEnumerable<XmlNode> credentials = credentialsXml?.ChildNodes[0]?.ChildNodes.Cast<XmlNode>();
+                foreach (XmlNode credential in credentials)
+                {
+                    string user = null;
+                    string password = null;
+                    foreach (XmlNode item in credential.ChildNodes.Cast<XmlNode>())
+                    {
+                        IEnumerable<XmlNode> current = item.Attributes.Cast<XmlNode>();
+                        if (current.Any(a => a.Value.Equals("username", StringComparison.OrdinalIgnoreCase)))
+                        {
+                            user = current.FirstOrDefault(a => a.Name.Equals("value", StringComparison.OrdinalIgnoreCase))?.Value;
+                        }
+
+                        if (current.Any(a => a.Value.Equals("password", StringComparison.OrdinalIgnoreCase) ||
+                                             a.Value.Equals("cleartextpassword", StringComparison.OrdinalIgnoreCase)))
+                        {
+                            password = current.FirstOrDefault(a => a.Name.Equals("value", StringComparison.OrdinalIgnoreCase))?.Value;
```

Comment:
>password [](http://example.com/codeflow?start=28&length=8)

so 'cleartextpassword' is plainly bad, but is 'password' encrypted? if we have an encrypted password will the validator ever fire or are we just generating fingerprint candidates that will never validate?

Reply:
In theory, ClearTextPassword would be the plaintext, but you could also save the same in the Password tag. Per my understanding:

when you use:
```
dotnet nuget add source [https://someServer/myTeam](https://someServer/myTeam) -n myTeam -u myUsername -p myPassword --store-password-in-clear-text
```

it would store as plaintext. If you don't use the store-password flag, it would encrypt it. But, you could manually
edit both and set the passwords. Still checking how to make an end-to-endd test